src/cmd/create.go

```diff
@@ -0,0 +1,490 @@
+/*
+Copyright © 2019 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/briandowns/spinner"
+
+	"github.com/containers/toolbox/pkg/podman"
+	"github.com/containers/toolbox/pkg/utils"
+
+	"github.com/godbus/dbus/v5"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	createFlags struct {
+		image   string
+		release string
+		unsafe  bool
+	}
+	ulimitHost               = []string{}
+	homeCanonical            = ""
+	homeLink                 = ""
+	mediaLink                = ""
+	mntLink                  = ""
+	toolboxProfileBind       = []string{}
+	sudoGroup                = ""
+	kcmSocket                = ""
+	kcmSocketBind            = []string{}
+	usrMountPoint            = ""
+	usrMountSourceFlags      = ""
+	usrMountDestinationFlags = "ro"
+	dbusSystemBusAddress     = ""
+	preservedEnvVars         = []string{
+		"COLORTERM",
+		"COLUMNS",
+		"DBUS_SESSION_BUS_ADDRESS",
+		"DBUS_SYSTEM_BUS_ADDRESS",
+		"DESKTOP_SESSION",
+		"DISPLAY",
+		"LANG",
+		"LINES",
+		"SSH_AUTH_SOCK",
+		"TERM",
+		"USER",
+		"VTE_VERSION",
+		"WAYLAND_DISPLAY",
+		"XDG_CURRENT_DESKTOP",
+		"XDG_DATA_DIRS",
+		"XDG_MENU_PREFIX",
+		"XDG_RUNTIME_DIR",
+		"XDG_SEAT",
+		"XDG_SESSION_DESKTOP",
+		"XDG_SESSION_ID",
+		"XDG_SESSION_TYPE",
+		"XDG_VTNR",
+	}
+)
+
+var createCmd = &cobra.Command{
+	Use:   "create [flags] NAME",
+	Short: "Create a new toolbox container",
+	Run: func(cmd *cobra.Command, args []string) {
+		create(args)
+	},
+	Args: cobra.MaximumNArgs(1),
+}
+
+func init() {
+	rootCmd.AddCommand(createCmd)
+
+	flags := createCmd.Flags()
+	flags.StringVarP(&createFlags.image, "image", "i", "", "Change the name of the base image used to create the toolbox container")
+	flags.StringVarP(&createFlags.release, "release", "r", "", "Create a toolbox container for a different operating system release than the host")
+	flags.BoolVar(&createFlags.unsafe, "unsafe", false, "Allow the use of a non-Toolbox-compliant image")
```

Comment:
I am curious about the `--unsafe`` flag? Was there a specific reason for it?

If really necessary it will be better to add it through a separate PR to keep the Go rewrite as conservative in scope as possible.

Reply:
People try to use non-toolbox images to create containers. It's not critical but something that can help with testing new images.

But surely we could separate it.